be/src/exprs/split_part.cpp

```diff
@@ -65,7 +66,17 @@ StatusOr<ColumnPtr> StringFunctions::split_part(FunctionContext* context, const
             if (part_number > haystack.size) {
                 res.append_null();
             } else {
-                res.append(Slice(haystack.data + part_number - 1, 1));
+                int char_size = 0, h = 0;
+                for (auto num = 0; h < haystack.size && num < part_number - 1; h += char_size) {
+                    char_size = UTF8_BYTE_LENGTH_TABLE[static_cast<unsigned char>(haystack.data[h])];
```

Comment:
does it affect performance ?

Reply:
It has to detect character size in O(n), besides it is a rare case where the delimiter is an empty string.